### ISSUE TITLE
No Reference Year in OHEA

### ISSUE THREAD

**Issue by daiClegg** [dbrownukk/EFD_v2#394]
OIHM Studies always have a reference year.
OHEA has no reference year. 
It does have a start date for the project.
Should it also have a reference year?

**Comment by celiapetty** [comment]
One for John to confirm, but my understanding is that the reference year should be recorded  Note: in HyCRISTAL (for example) the Project Start Date is 2016-ie when the project started....The baseline/reference year will be an earlier date. This is identified in the preliminary community interviews when a  timeline is drawn up that identifies the baseline-reference year (ie  the timeline of good years eg 2010, 2013, bad years 2011, 2012, neither good nor bad ie baseline-reference year 2014. Timelines usually cover the past 8-10 years). It should  be mandatory to upload this contextual data, along with other information eg seasonal calendars, market prices, that are part of the HEA field study protocol. The facility to upload this data is there-I wonder if a red flag could be raised if the context data isn't uploaded. Also for information, baseline/reference years have been used interchangeably. John-we should make a rule for O-HEA. Which term do you prefer?

**Comment by daiClegg** [comment]
And, as per ourt discussion re refernece year in OIHM, the format would be dd/mm/yy and the prompt would be 'Reference Year Start Date:'

**Comment by daiClegg** [comment]
@DrJohnSeaman any thoughts about this? Happy to leave it as it is, but having noticed the difference i just wanted to check it was intended.

**Comment by DrJohnSeaman** [comment]
Happy New Year

OHEA should have a reference year.  Start and end dates .

Celia's suggestion of a flag for contextual data is a good one.

**Comment by daiClegg** [comment]
Isn't the end date 12 months after the start date?

**Comment by daiClegg** [comment]
Issue #331 is where the reference year is discussed wrt OIHM. I propose we do the same for the 2.

**Comment by daiClegg** [comment]
@celiapetty , @DrJohnSeaman 

> I wonder if a red flag could be raised if the context data isn't uploaded.

What would you want the app to do, and at what point in the process?
And how is the app to validate that the 'context data' acceptable?

Right now you can upload any files to attach to a project or study (and to some other data elements), but the app doesn't care what they are, nor does it block progress if no files are loaded ay any point in the process.

**Comment by celiapetty** [comment]
How about: _Have you uploaded the following - ....... If  there are any gaps, please email the study director (currently this would be John or me)_  We need to agree the minimum set of HEA and IHM context data eg to include GIS co-ordinates of all villages surveyed; sketch map of LZ/s; seasonal calendars; crop and livestock prices in study year; employment rates; wild foods available plus other relevant secondary data available electronically eg district development plans. We have standard context information forms for IHM that would be uploaded here. Thoughts? Also-the contracts of IHM study leads includes a requirement to forward all contextual information to EfD. This could be included in future HEA contracts.

**Comment by daiClegg** [comment]
We would need to know how the app is to evaluate whether there is missing data, and at what point it should do that because right now there is no point at which the app can 'know' the study is complete.
Also there is no way the app can know, for example,  that any of the attached files is a sketch map of the LZ.

**Comment by celiapetty** [comment]
Is there any way of introducing a question: Has context data been uploaded: yes/no ?
In any event, we will include instructions in guidelines for management of field studies.

**Comment by daiClegg** [comment]
We could add a yes/no field called Context Data Uploaded on the Project record,
It would default to No. 
It would not be valid to change it to Yes unless at least one file had been uploaded to the project.

The questions are, when do you want the software to care that it's not Yes? 

- Any time you look at any record in the project? 
- Any time you look at that Project record itself? 
- Any time you run a report on any data in that project? 

And what do you want it to do? 

- Log the user out? 
- Throw an error if they try to do anything apart from upload a file to the notes & change No to Yes?

There has to be a defined trigger, and it has to be at the point when the user should have done it and you want to freeze them out unless they do it now.

I don't think there is any such point. I might have a deadline to produce my project report by tomorrow; i've got all my data, done all my analysis, corrected any errors and i just need to run the final reports to paste the charts into my report. But i don't have the laptop that has the context data notes with me. Do you want to stop me?

**Comment by celiapetty** [comment]
Agreed. This sounds like a bad idea so we can drop it.  For information (although not a programming issue) - Currently, IHM Service TOR for  field study leads, (appended to contracts) include a requirement that all survey and contextual data must be correctly filed and forwarded to EfD at the end of the field study. This means final payment isn't made until electronic copies of everything have been received. The approach could be adapted for OHEA and OIHM contracts where EfD is involved and will be described in  training protocols for field study management . Other organisations using OHEA and OIHM should then at least know what best practice looks like.
